routing/pathfind.go

```diff
@@ -752,11 +759,24 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 
 			// If the node exists and has valid features, use them.
 			case err == nil:
-				err := feature.ValidateDeps(targetNode.Features)
-				if err == nil {
-					fromFeatures = targetNode.Features
+				nodeFeatures := targetNode.Features
+
+				// Don't route through nodes that contain
+				// unknown required features.
+				req := nodeFeatures.UnknownRequiredFeatures()
+				if len(req) > 0 {
+					break
 				}
```

Comment:
Non blocking suggestion: create `feature.ValidateRequired` that returns an error describing the missing features. This can then be used in three locations where we currently check the length of the returned list.

Reply:
i also thought this, and since you've brought it up i decided to implement. fixed